src/drivers/amd/rembrandt/acp_dmic_dai.c

```diff
@@ -120,7 +135,15 @@ static int acp_dmic_dai_get_hw_params(struct dai *dai,
 		dai_info(dai, "unsupported samplerate %d", acpdata->dmic_params.pdm_rate);
 		params->rate = ACP_DEFAULT_SAMPLE_RATE;
 	}
-	params->channels = ACP_DEFAULT_NUM_CHANNELS;
+	switch (acpdata->dmic_params.pdm_ch) {
+	case 2:
+	case 4:
+		params->channels = acpdata->dmic_params.pdm_ch;
+		break;
+	default:
+		dai_info(dai, "unsupported channels %d", acpdata->dmic_params.pdm_ch);
```

Comment:
should this be a warning or an error? Similar in other such "unsupported" locations

Reply:
changes done